packages/application/src/shell.ts

```diff
@@ -57,10 +59,115 @@ const MODE_ATTRIBUTE = 'data-shell-mode';
 
 const ACTIVITY_CLASS = 'jp-Activity';
 
+/* tslint:disable */
+/**
+ * The layout restorer token.
+ */
+export const ILabShell = new Token<ILabShell>(
+  '@jupyterlab/application:ILabShell'
+);
+/* tslint:enable */
+
+/**
+ * The JupyterLab application shell.
+ */
+export interface ILabShell extends LabShell {}
+
+/**
+ * The namespace for `ILabShell` type information.
+ */
+export namespace ILabShell {
+  /**
+   * The areas of the application shell where widgets can reside.
+   */
+  export type Area = 'main' | 'top' | 'left' | 'right' | 'bottom';
+
+  /**
+   * The restorable description of an area within the main dock panel.
+   */
+  export type AreaConfig = DockLayout.AreaConfig;
+
+  /**
+   * An arguments object for the changed signals.
+   */
+  export type IChangedArgs = FocusTracker.IChangedArgs<Widget>;
+
+  /**
+   * A description of the application's user interface layout.
+   */
+  export interface ILayout {
+    /**
+     * Indicates whether fetched session restore data was actually retrieved
+     * from the state database or whether it is a fresh blank slate.
+     *
+     * #### Notes
+     * This attribute is only relevant when the layout data is retrieved via a
+     * `fetch` call. If it is set when being passed into `save`, it will be
+     * ignored.
+     */
+    readonly fresh?: boolean;
+
+    /**
+     * The main area of the user interface.
+     */
+    readonly mainArea: IMainArea | null;
+
+    /**
+     * The left area of the user interface.
+     */
+    readonly leftArea: ISideArea | null;
+
+    /**
+     * The right area of the user interface.
+     */
+    readonly rightArea: ISideArea | null;
```

Comment:
Should there also be entries here for top and bottom areas, even if they don't have as rich of APIs?

Reply:
Sounds like an "enhancement" and "help wanted" to me 😁

Reply:
Ayup